src/app/CommandHandler.cpp

```diff
@@ -911,6 +911,112 @@ void CommandHandler::MoveToState(const State aTargetState)
     ChipLogDetail(DataManagement, "Command handler moving to [%10.10s]", GetStateStr());
 }
 
+#if CHIP_WITH_NLFAULTINJECTION
+
+namespace {
+
+CHIP_ERROR TestOnlyExtractCommandPathFromNextInvokeRequest(TLV::TLVReader & invokeRequestsReader,
+                                                           ConcreteCommandPath & concretePath)
+{
+    CommandPathIB::Parser commandPath;
+    ReturnErrorOnFailure(invokeRequestsReader.Next());
+    VerifyOrReturnError(TLV::AnonymousTag() == invokeRequestsReader.GetTag(), CHIP_ERROR_INVALID_ARGUMENT);
+    CommandDataIB::Parser commandData;
+    ReturnErrorOnFailure(commandData.Init(invokeRequestsReader));
+    ReturnErrorOnFailure(commandData.GetPath(&commandPath));
+    return commandPath.GetConcreteCommandPath(concretePath);
+}
+
+} // anonymous namespace
+
+// This method copies code from other sections of this file. Minimal attempt is made at consolidating
+// repetitive code because within this method we want the server to crash with a helpful message that
+// is actionable for understanding why the test failed.
+void CommandHandler::TestOnlyTcIdm1_3FaultInjection(Messaging::ExchangeContext * ec, System::PacketBufferHandle && payload,
+                                                    bool isTimedInvoke, bool oneReponsePerMessage, bool invertResponseOrdering,
```

Comment:
bools are hard to read when viewing calls. 4 bools in a row are extra hard. Could we make a wrapper struct/builder or bitmask or something else?